File: Equal Left and Right Subarray Sum.cpp
int equalSum(int N, vector<int> &A) {
        // code here
        long long ls=0,rs=0,ans=-1,i=0;
        int s=0,e=N-1;
        while(s<=e){
            if(ls==rs && i<N-1){
                ls+=A[s++];
                rs+=A[e--];
                i+=2;
            }
            else if(ls>rs) rs+=A[e--],i++;
            else if(ls<rs) ls+=A[s++],i++;
            else if(ls==rs && i==N-1){
                ans=s+1;
                break;
            }
            
        }
       
       return ans;
    }

File: Equal Left and Right Subarray Sum.java
class Solution{
	int equalSum(int [] A, int N) {
		//Write your code here
		int after = 0, before = 0;
		
		for(int n : A) {
		    after += n;
		}
		
		for(int i = 0; i< N; i++) {
		    if(i > 0){
		        before += A[i-1];
		    }
		    after -= A[i];
		    if(after == before) {
		        return i+1;  // according to question array is one indexed
		    }
		}
		return -1;
	}
}
